File: Backend/app/dtos/users/admin_response_dto.py

from app.dtos.common.base_response_dto import BaseResponseDTO
from app.dtos.users.admin_user_dto import AdminCreateUserDataDTO, AdminUpdateUserDataDTO, AdminDeleteUserDataDTO, AdminFindUserDataDTO, AdminFindUserDataListDTO


class AdminCreateUserResponseDTO(BaseResponseDTO[AdminCreateUserDataDTO]):
    pass

class AdminUpdateUserResponseDTO(BaseResponseDTO[AdminUpdateUserDataDTO]):
    pass

class AdminDeleteUserResponseDTO(BaseResponseDTO[AdminDeleteUserDataDTO]):
    pass

class AdminFindUserResponseDTO(BaseResponseDTO[AdminFindUserDataDTO]):
    pass


class AdminFindUserResponseDTOList(BaseResponseDTO[AdminFindUserDataListDTO]):
    pass

File: Frontend/src/types/dtos/users/admin_response_dto.ts
import type { ApiResponse } from "~/types/common/api-response.type";
import { mapAliasList, mapAliases, type AliasMap } from "~/utils";
import type {
  AdminCreateUserDataDTO,
  AdminUpdateUserDataDTO,
  AdminDeleteUserDataDTO,
  AdminFindUserDataDTO,
} from "./admin_user_dto";


export type AdminCreateUserResponseDTO = ApiResponse<AdminCreateUserDataDTO>;

export type AdminUpdateUserResponseDTO = ApiResponse<AdminUpdateUserDataDTO>;

export type AdminDeleteUserResponseDTO = ApiResponse<AdminDeleteUserDataDTO>;

export type AdminFindUserResponseDTO = ApiResponse<AdminFindUserDataDTO>;

export type AdminFindUserResponseDTOList = ApiResponse<AdminFindUserDataDTO[]>;
